Fitness/Controllers/HomeController.swift

```diff
@@ -317,63 +361,68 @@ extension HomeController: UICollectionViewDelegate, UICollectionViewDelegateFlow
             navigationController?.pushViewController(proDetailViewController, animated: true)
             return
         }
-
+        
         switch sections[indexPath.section] {
-        case .allGyms:
-            let gymDetailViewController = GymDetailViewController()
-            gymDetailViewController.gym = gyms[indexPath.row]
-            navigationController?.pushViewController(gymDetailViewController, animated: true)
-        case .lookingFor:
-            let cal = Calendar.current
-            let currDate = Date()
-            guard let startDate = cal.date(bySettingHour: 0, minute: 0, second: 0, of: currDate) else { return }
-            let endDate = cal.date(bySettingHour: 23, minute: 59, second: 0, of: currDate) ?? Date()
-
-            let filterParameters = FilterParameters(applyFilter: true, startingTime: startDate,  endingTime: endDate, instructorsNames: [], classesNames: [], gymsIds: [], tagsNames: [tags[indexPath.row].name])
-
-            guard let classNavigationController = tabBarController?.viewControllers?[1] as? UINavigationController else { return }
-            guard let classListViewController = classNavigationController.viewControllers[0] as? ClassListViewController else { return }
-
-            classListViewController.currentFilterParams = filterParameters
-            classNavigationController.setViewControllers([classListViewController], animated: false)
-
-            tabBarController?.selectedIndex = 1
+        case .checkIns:
+            if habits.count == 0 {
+                pushHabitOnboarding()
+            }
+            return
         case .pros:
             let proDetailViewController = ProBioPageViewController()
             navigationController?.hidesBottomBarWhenPushed = true
             navigationController?.pushViewController(proDetailViewController, animated: true)
-        case .todaysClasses:
-            let classDetailViewController = ClassDetailViewController()
-            classDetailViewController.gymClassInstance = gymClassInstances[indexPath.row]
-            navigationController?.hidesBottomBarWhenPushed = true
-            navigationController?.pushViewController(classDetailViewController, animated: true)
+        case .todaysClasses, .allGyms:
+            return
         }
-
-        // MARK: - Fabric
-        Answers.logCustomEvent(withName: "Found Info on Homepage", customAttributes: [
-            "Section": sections[indexPath.section].rawValue
-            ])
     }
 }
 
-// MARK: - NavigationDelegate
-extension HomeController: NavigationDelegate {
+extension HomeController: ChooseGymsDelegate {
+    func updateFavorites(favorites: [String]) {
+        favoriteGyms = []
+        
+        for favoriteGym in favorites {
+            for gym in self.gyms {
+                if gym.name == favoriteGym {
+                    self.favoriteGyms.append(gym)
+                    continue
+                }
+            }
+        }
```

Comment:
i think we can do this in a shorter way:
```
favoriteGyms = favorites.compactMap { favorite in 
  self.gyms.first { $0.name == favorite } 
}
```
syntax might be a bit off so you'll have to double check that

Reply:
Will do! much swiftier